src/macros.rs

```diff
@@ -223,3 +223,47 @@ macro_rules! check_piece_count {
         }
     }}
 }
+
+macro_rules! polynom {
+    [$value:expr; $count:expr] => {{
+        let array = [$value; $count];
+        $crate::poly::Polynom::from(&array[..])
+    }}; 
+
+    [$( $value:expr ),* ] => {{
+        let array = [$($value, )*];
+        $crate::poly::Polynom::from(&array[..])
+    }};
+}
+
+macro_rules! uncheck {
+    ($array:ident[$index:expr]) => {
+        if cfg!(feature = "unsafe_indexing") {
+            unsafe {
+                *$array.get_unchecked($index)
+            }
+        } else {
+            $array[$index]
+        }
+    }
+}
+
+macro_rules! uncheck_mut {
+    ($array:ident[$index:expr]) => {
+        * if cfg!(feature = "unsafe_indexing") {
+            unsafe {
```

Comment:
I would recommend removing the unsafe indexing entirely. rustc is good at eliding bounds checks where possible